src/lib/message/L1ToL2Message.ts

```diff
@@ -222,24 +225,44 @@ export class L1ToL2MessageReader extends L1ToL2Message {
    * May throw an error if retryable ticket has yet to be created
    * @returns
    */
-  public getRetryableCreationReceipt(): Promise<TransactionReceipt> {
-    return this.l2Provider.getTransactionReceipt(this.retryableCreationId)
+  public async getRetryableCreationReceipt(    
+    confirmations?: number,
+    timeout?: number
+  ): Promise<TransactionReceipt> {
+    if (!this.retryableCreationReceipt){
+      if (timeout){
+        this.retryableCreationReceipt = await this.l2Provider.waitForTransaction(this.retryableCreationId, confirmations, timeout)
+      } else {
+        this.retryableCreationReceipt = await this.l2Provider.getTransactionReceipt(this.retryableCreationId)
+      }
+    }
+    return this.retryableCreationReceipt
   }
 
   // CHRIS: TODO: update these docs
-  /**
-   * Receipt for the l2 transaction created by this message. See L1ToL2Message.l2TxHash
+    /**
+   * Receipt for the successful l2 transaction created by this message. See L1ToL2Message.l2TxHash
    * May throw an error if the l2 transaction has yet to be executed, which is the case if
    * the retryable ticket has not been created and redeemed.
-   * @returns
+   * @returns TransactionReceipt of the first successful redeem if exists, otherwise null
+   */
+  public async getSuccessfulRedeem(): Promise<TransactionReceipt | null> {
+    return await this.getFirstSuccessfulRedeem()
+  }
+
+    /**
+   * Receipt for the auto-redeem l2 transaction created by this message. See L1ToL2Message.l2TxHash
+   * May throw an error if the l2 transaction has yet to be executed, which is the case if
+   * the retryable ticket has not been created and redeemed.
+   * @returns TransactionReceipt of the auto redeem attempt if exists, otherwise null
    */
-  public async getFirstRedeemAttempt(): Promise<TransactionReceipt | null> {
-    return await this.getFirstRedeem()
+  public async getAutoRedeemAttempt(): Promise<TransactionReceipt | null> {
+    return await this.getAutoRedeem()
   }
 
   // CHRIS: TODO: read ALL the comments in arbitrum sdk and check for accuracy with nitro
 
-  private async getFirstRedeem(
+  private async getAutoRedeem(
```

Comment:
I think we can get rid of the `retryableCreationReceipt?: TransactionReceipt` arg (since we can use the internal cached method). Then the public and private functions will have the same signature, so I think can just merge them / get rid of the private variant